orion/pkg/controllers/session_controller.go

```diff
@@ -35,21 +37,29 @@ func GetSessionById(c *gin.Context) {
 	return
 }
 
-func CreateSession(c *gin.Context) {
+func CreateSessions(c *gin.Context) {
 	// Incoming JSON
-	var sessionJson domains.Session
-	c.BindJSON(&sessionJson)
+	var sessionsJson, validatedSessions []domains.Session
+	c.BindJSON(&sessionsJson)
 
-	if err := sessionJson.Validate(); err != nil {
-		c.Error(err)
-		c.String(http.StatusBadRequest, err.Error())
-		return
+	var errorString string
+	for _, session := range sessionsJson {
+		if err := session.Validate(); err != nil {
+			errorString = appendError(errorString, fmt.Sprint(session.Id), err)
+		} else {
+			validatedSessions = append(validatedSessions, session)
+		}
 	}
```

Comment:
Actually now that I think about it, would it be cleaner to move validation to the repo layer as well instead of iterating over the list of sessions twice?

Reply:
I think that's fair. Make sure to call the validation function before trying to persist to db (Calling the Exec statement).